apps/cow-tools/src/app/milkman/(components)/orderForm/Overview.tsx

```diff
@@ -0,0 +1,217 @@
+import { formatDateToLocalDatetime } from "@bleu-fi/utils/date";
+import { formatNumber } from "@bleu-fi/utils/formatNumber";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { ExclamationTriangleIcon } from "@radix-ui/react-icons";
+import { useEffect, useState } from "react";
+import { FieldValues, useForm } from "react-hook-form";
+import { formatUnits } from "viem";
+
+import { TokenSelect } from "#/app/milkman/(components)/TokenSelect";
+import { TransactionStatus } from "#/app/milkman/utils/type";
+import { Checkbox } from "#/components/Checkbox";
+import { Input } from "#/components/Input";
+import { Tooltip } from "#/components/Tooltip";
+import { Form, FormMessage } from "#/components/ui/form";
+import { useSafeBalances } from "#/hooks/useSafeBalances";
+import { orderOverviewSchema } from "#/lib/schema";
+
+import { tokenPriceChecker } from "../../[network]/order/new/page";
+import { FormFooter } from "./Footer";
+
+export function FormOrderOverview({
+  onSubmit,
+  userAddress,
+  defaultValues,
+}: {
+  onSubmit: (data: FieldValues) => void;
+  userAddress: string;
+  defaultValues?: FieldValues;
+}) {
+  const form = useForm<typeof orderOverviewSchema._type>({
+    resolver: zodResolver(orderOverviewSchema),
+    mode: "onSubmit",
+  });
+  const {
+    register,
+    setValue,
+    clearErrors,
+    formState: { errors },
+    watch,
+  } = form;
+
+  useEffect(() => {
+    register("tokenBuy");
+    register("tokenSell");
+    register("validFrom");
+  }, []);
+
+  const { assets, loaded } = useSafeBalances();
+
+  const formData = watch();
+  const tokenSell = assets.find(
+    (asset) => asset.tokenInfo.address === formData.tokenSell?.address,
+  );
+
+  const walletAmount = !tokenSell
+    ? 0
+    : formatUnits(BigInt(tokenSell?.balance), tokenSell?.tokenInfo.decimals);
+
+  const [isValidFromNeeded, setIsValidFromNeeded] = useState(
+    !!defaultValues?.validFrom,
+  );
+
+  useEffect(() => {
+    if (
+      loaded &&
+      defaultValues?.tokenBuy?.address &&
+      defaultValues?.tokenSell?.address
+    ) {
+      setValue("tokenBuy", defaultValues?.tokenBuy);
+      setValue("tokenSell", defaultValues?.tokenSell);
+    }
+  }, [loaded]);
+
+  useEffect(() => {
+    if (isValidFromNeeded === false) {
+      setValue("validFrom", undefined);
+      clearErrors("validFrom");
+    }
+  }, [isValidFromNeeded]);
```

Comment:
is this needed? thought isValidFromNeeded as a form attribute and just not use it on onSubmit instead of using as a state